ui/src/app/views/workspace-edit/component.html

```diff
@@ -118,6 +119,24 @@ <h3 class="modal-title">Error:</h3>
   <div class="modal-body">Could not create workspace.</div>
   <div class="modal-footer">
     <button type="button" class="btn btn-primary" (click)="navigateBack()">Cancel Creation</button>
-    <button type="button" class="btn btn-primary" (click)="resetWorkspaceCreation()">Keep Editing</button>
+    <button type="button" class="btn btn-primary" (click)="resetWorkspaceEditor()">Keep Editing</button>
+  </div>
+</clr-modal>
+<clr-modal [(clrModalOpen)]="workspaceUpdateError">
+  <h3 class="modal-title">Error:</h3>
+  <div class="modal-body">Could not update workspace.</div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-primary" (click)="navigateBack()">Cancel Edits</button>
+    <button type="button" class="btn btn-primary" (click)="resetWorkspaceEditor()">Keep Editing</button>
+  </div>
+</clr-modal>
+<clr-modal [(clrModalOpen)]="workspaceUpdateConflictError">
+  <h3 class="modal-title">Conflicting update:</h3>
+  <div class="modal-body">
+    Workspace has been modified since the beginning of this editing session.
+  </div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-primary" (click)="reloadConflictingWorkspace()">Reload Workspace</button>
+    <button type="button" class="btn btn-primary" (click)="resetWorkspaceEditor()">Keep Editing</button>
```

Comment:
Do we want to allow the user to keep editing without reloading the conflicting workspace? I'm a little unsure what the expected behavior here should be. I would probably remove the `Keep Editing` button, and instead have an `Overwrite` button or something?

Reply:
Good question. I also wasn't too sure. I think we should go with 2 buttons probably to avoid overwhelming the user - in most cases, I'd suspect they were just making a minor tweak and they would be fine just reloading. In other cases, they may have typed an essay into the description, so they'll want to effectively just close the dialog and copy their work before reloading (at least that's what I would do).

In general, I think it should be rare that the user would see this dialog, so I think my preference would be to provide the bare minimum idiot-proofed options to the user (basically what's in this PR currently). Overwrite could get hairy, depending on what the scope of this overwrite is, i.e. we may need to add some more complex logic in applying the update.

I think I'm proposing leave it as currently is in the PR. If the product flow changes such that this is a frequently seen dialog for some reason, I think we should consider revisiting this, definitely agree that it could be improved. How does this sound?